### PR TITLE
ENG-8554: Enable TestCommandLog JUnit test and isolate more DRv1 logic

### PR DESCRIPTION
**Pull request by wweiss-voltdb**

### CODE REVIEW

File: src/frontend/org/voltcore/messaging/TransactionInfoBaseMessage.java [hunk 1]
```diff
@@ -139,7 +139,7 @@ public boolean isForReplay() {
         return m_isForReplay;
     }
 
-    public boolean isForDR() {
```
Comment (nshi) [inline src/frontend/org/voltcore/messaging/TransactionInfoBaseMessage.java]: We can probably get rid of this API.

File: src/frontend/org/voltdb/iv2/SpProcedureTask.java [hunk 2]
```diff
@@ -50,7 +50,7 @@
         HOST_TRACE_ENABLED = hostLog.isTraceEnabled();
     }
 
-    SpProcedureTask(Mailbox initiator, String procName, TransactionTaskQueue queue,
+    public SpProcedureTask(Mailbox initiator, String procName, TransactionTaskQueue queue,
```
Comment (nshi) [inline src/frontend/org/voltdb/iv2/SpProcedureTask.java]: This seems like an unnecessary change.

Comment (wweiss-voltdb) [inline src/frontend/org/voltdb/iv2/SpProcedureTask.java]: It needs to be public because the TestCommandLog is not part of the iv2 package.

File: src/frontend/org/voltcore/messaging/TransactionInfoBaseMessage.java (hunk 1 above)
Comment (wweiss-voltdb) [inline src/frontend/org/voltcore/messaging/TransactionInfoBaseMessage.java]: I agree. I came across it and wanted to make sure we did not use it with an assert for a couple of weeks before removing it.

**Comment by nshi** [comment]
Merged externally
